wp-includes/widgets/class-wp-widget-media-gallery.php

```diff
@@ -0,0 +1,231 @@
+					'default' => 3,
+				),
+				'size' => array(
+					'type' => 'string',
+					'enum' => array_merge( get_intermediate_image_sizes(), array( 'full', 'custom' ) ),
+					'default' => 'thumbnail',
+				),
+				'link_type' => array(
+					'type' => 'string',
+					'enum' => array( 'none', 'file', 'post' ),
+					'default' => 'none',
+					'media_prop' => 'link',
+					'should_preview_update' => false,
+				),
+				'orderby_random' => array(
+					'type'                  => 'boolean',
+					'default'               => false,
+					'media_prop'            => '_orderbyRandom',
+					'should_preview_update' => false,
+				),
+				'attachments' => array(
+					'type'                  => 'string',
+					'default' => '',
+				),
+			)
+		);
+	}
+
+	/**
+	 * Render the media on the frontend.
+	 *
+	 * @since  4.8.0
+	 * @access public
+	 *
+	 * @param array $instance Widget instance props.
+	 * @return void
+	 */
+	public function render_media( $instance ) {
+		$instance = array_merge( wp_list_pluck( $this->get_instance_schema(), 'default' ), $instance );
+		$shortcode_atts = array(
+			'ids' => $instance['ids'],
+		);
+		// @codingStandardsIgnoreStart
+		if ( $instance['orderby_random'] ) {
+			$shortcode_atts['orderby'] = 'rand';
+		}
+		// @codingStandardsIgnoreEnd
+
+		echo gallery_shortcode( $shortcode_atts );
+	}
+
+	/**
+	 * Loads the required media files for the media manager and scripts for media widgets.
+	 *
+	 * @since 4.8.0
+	 * @access public
+	 */
+	public function enqueue_admin_scripts() {
+		parent::enqueue_admin_scripts();
+
+		$handle = 'media-gallery-widget';
+		wp_enqueue_script( $handle );
+
+		$exported_schema = array();
+		foreach ( $this->get_instance_schema() as $field => $field_schema ) {
+			$exported_schema[ $field ] = wp_array_slice_assoc( $field_schema, array( 'type', 'default', 'enum', 'minimum', 'format', 'media_prop', 'should_preview_update' ) );
+		}
+		wp_add_inline_script(
+			$handle,
+			sprintf(
+				'wp.mediaWidgets.modelConstructors[ %s ].prototype.schema = %s;',
+				wp_json_encode( $this->id_base ),
+				wp_json_encode( $exported_schema )
+			)
+		);
+
+		wp_add_inline_script(
+			$handle,
+			sprintf(
+				'
+					wp.mediaWidgets.controlConstructors[ %1$s ].prototype.mime_type = %2$s;
+					_.extend( wp.mediaWidgets.controlConstructors[ %1$s ].prototype.l10n, %3$s );
+				',
+				wp_json_encode( $this->id_base ),
+				wp_json_encode( $this->widget_options['mime_type'] ),
+				wp_json_encode( $this->l10n )
+			)
+		);
+	}
+
+	/**
+	 * Render form template scripts.
+	 *
+	 * @since 4.8.0
+	 * @access public
+	 */
+	public function render_control_template_scripts() {
+		parent::render_control_template_scripts();
+		?>
+		<script type="text/html" id="tmpl-wp-media-widget-gallery-preview">
+			<# var describedById = 'describedBy-' + String( Math.random() ); #>
+			<# data.attachments = data.attachments ? JSON.parse(data.attachments) : ''; #>
+			<# if ( data.error && 'missing_attachment' === data.error ) { #>
```

Comment:
Looks like we never set the `missing_attachment` error in this widget so we could likely remove this logic.